explorer/explorer.go

```diff
@@ -41,18 +94,71 @@ func ListenEvents(event event.Event) {
 //
 // In some platforms the resulting `io.ReadCloser` is a `os.File`, but it's not
 // a guarantee.
-func ReadFile(extensions ...string) (io.ReadCloser, error) {
-	return openFile(extensions...)
+//
+// It's a blocking call, you should call it on a separated goroutine. For most OSes, only one
+// Import or Export, can happen at the same time, for each app.Window/Explorer.
+func (e *Explorer) Import(extensions ...string) (io.ReadCloser, error) {
+	if e == nil {
+		return nil, ErrNotAvailable
+	}
+
+	if runtime.GOOS != "js" {
+		e.mutex.Lock()
+		defer e.mutex.Unlock()
+	}
+
+	return e.importFile(extensions...)
 }
 
-// WriteFile opens the file selector, and writes the given content into
+// Export opens the file selector, and writes the given content into
 // some file, which the use can choose the location.
 //
 // It's important to close the `io.WriteCloser`. In some platforms the
 // file will be saved only when the writer is closer.
 //
-// In some platforms the resulting `io.ReadCloser` is a `os.File`, but it's not
+// In some platforms the resulting `io.WriteCloser` is a `os.File`, but it's not
 // a guarantee.
+//
+// It's a blocking call, you should call it on a separated goroutine. For most OSes, only one
+// Import or Export, can happen at the same time, for each app.Window/Explorer.
+func (e *Explorer) Export(name string) (io.WriteCloser, error) {
```

Comment:
What motivated the name change from `WriteFile` to `Export`?

Reply:
This conversation is happening over [here](https://github.com/gioui/gio-x/pull/6#discussion_r775921004)